### PR TITLE
Debate comment credentials

### LINKED ISSUE #131: Replace debate nickname with first name and last name
Do not require nickname on landing page, ask for first name and last name when adding first comment, cannot change first name and last name for one debate

### LINKED ISSUE #134: Remove nickname from debate login screen

### PR DESCRIPTION
**Pull request by jozzi05**
Closes #131 
Closes #134

### CODE REVIEW

**Comment by jozzi05** [comment]
Do not merge till new api.

File: el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt [hunk 1]
```diff
@@ -5,25 +5,30 @@ import com.elpassion.android.commons.sharedpreferences.createSharedPrefs
 import com.elpassion.sharedpreferences.gsonadapter.gsonConverterAdapter
```
Comment (macPry) [inline el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt]: |`line 16`|
|---|
Maybe use simpler form like: areCredentialsAvailable

File: el-debate/src/androidTest/java/pl/elpassion/elspace/debate/comment/DebateCommentActivityTest.kt [hunk 2]
```diff
@@ -113,48 +108,37 @@ class DebateCommentActivityTest {
...
         onText(message).isDisplayed()
     }
 
     @Test
     fun shouldShowLoaderOnSendComment() {
         startActivity()
-        onId(R.id.debateCommentInputText)
-                .replaceText("message")
-                .pressImeActionButton()
+        sendMessage("message")
```
Comment (macPry) [inline el-debate/src/androidTest/java/pl/elpassion/elspace/debate/comment/DebateCommentActivityTest.kt]: |`line 224`|
|---|
Use default value for message

File: el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt (hunk 1 above)
Comment (jozzi05) [inline el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt]: What do you mean?
areCredentialsAvailable =!areCredentialsMissing

File: el-debate/src/androidTest/java/pl/elpassion/elspace/debate/comment/DebateCommentActivityTest.kt (hunk 2 above)
Comment (jozzi05) [inline el-debate/src/androidTest/java/pl/elpassion/elspace/debate/comment/DebateCommentActivityTest.kt]: :+1:

**Comment by macPry** [comment]
|`el-debate/src/main/java/pl/elpassion/elspace/debate/comment/DebateCommentController.kt:43`|
|---|
I would use switch:
when {
credentials.lastName.isBlank()->view.showLastNameError()
credentials.firstName.isBlank()->view.showFirstNameError()
else->{ debateRepo.saveTokenCredentials(token, credentials); view.closeCredentialsDialog() }
}

**Comment by jozzi05** [comment]
@macPry change it to when and then check if tests will pass.

File: el-debate/src/main/java/pl/elpassion/elspace/debate/login/DebateLoginController.kt [hunk 3]
```diff
@@ -15,30 +15,21 @@ class DebateLoginController(
...
-                view.fillDebateNickname(it)
-            }
+        debateRepo.getLatestDebateCode()?.let {
+            view.fillDebateCode(it)
         }
     }
 
-    fun onLogToDebate(debateCode: String, nickname: String) {
+    fun onLogToDebate(debateCode: String) {
         when {
```
Comment (macPry) [inline el-debate/src/main/java/pl/elpassion/elspace/debate/login/DebateLoginController.kt]: if - else would be better in this case

File: el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt (hunk 1 above)
Comment (macPry) [inline el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt]: fun areCredentialsAvailable(token: String): Boolean
and impl:
override fun areCredentialsAvailable(token: String): Boolean = credentialsRepository.read(token) != null

**Comment by macPry** [comment]
|`el-debate/src/test/java/pl/elpassion/elspace/debate/comment/DebateCommentControllerTest.kt:209`|
|---|
Rename to shouldShowLastNameErrorOnBlankLastName

File: el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt (hunk 1 above)
Comment (jozzi05) [inline el-debate/src/main/java/pl/elpassion/elspace/debate/DebatesRepository.kt]: Find place where this method is being used and then explain why should I invert this boolean.

**Comment by macPry** [comment]
|`el-debate/src/test/java/pl/elpassion/elspace/debate/comment/DebateCommentControllerTest.kt:223`|
|---|
2x verify in single test;)

**Comment by jozzi05** [comment]
It is by design. This test force me to implement business logic in the way that two errors are being shown, on incorrect inputs.

File: el-debate/src/main/java/pl/elpassion/elspace/api/RetrofitProvider.kt [hunk 4]
```diff
@@ -21,6 +22,14 @@ private val httpLoggingInterceptor by lazy {
     HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)
 }
 
+private val apiVersionHeaderInterceptor = Interceptor { chain ->
+    val originalRequest = chain.request()
+    val request = originalRequest.newBuilder()
+            .header("Accept", "edge=1")
+            .method(originalRequest.method(), originalRequest.body())
```
Comment (ghostbuster91) [inline el-debate/src/main/java/pl/elpassion/elspace/api/RetrofitProvider.kt]: Redundant line

**Pull request merged**